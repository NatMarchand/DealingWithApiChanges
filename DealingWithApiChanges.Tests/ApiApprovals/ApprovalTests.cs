using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using DealingWithApiChanges.MyApi;
using Xunit;

[assembly: UseReporter(typeof(XUnit2Reporter))]
namespace DealingWithApiChanges.Tests.ApiApprovals
{
    public class ApprovalTests
    {
        [Fact]
        public void TestApi()
        {
            var publicApiSnapshot = PublicApiGenerator.ApiGenerator.GeneratePublicApi(typeof(Person).Assembly);
            Approvals.Verify(publicApiSnapshot);
        }
    }
}
