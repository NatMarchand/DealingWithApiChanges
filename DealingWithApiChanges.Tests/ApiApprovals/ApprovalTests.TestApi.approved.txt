[assembly: System.Runtime.InteropServices.ComVisibleAttribute(false)]
[assembly: System.Runtime.InteropServices.GuidAttribute("0ca4a0d1-e909-4871-8440-652cea687fd8")]
[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.7.1", FrameworkDisplayName=".NET Framework 4.7.1")]
namespace DealingWithApiChanges.MyApi
{
    public interface ICanGreet
    {
        void SayHello();
    }
    public class Person : DealingWithApiChanges.MyApi.ICanGreet
    {
        public Person() { }
        public string Name { get; set; }
        public void SayHello() { }
    }
}